src/main/java/es/andrewazor/demo/App.java

```diff
@@ -15,7 +15,9 @@
     public void start() {
         getVertx().createHttpServer(
                 new HttpServerOptions()
-                .setPort(8080)
+                .setPort(
+                    System.getenv("HTTP_PORT")
+                    )
```

Comment:
```bash
$ ./gradlew build

> Task :compileJava
/home/andrew/workspace/vertx-fib-demo/src/main/java/es/andrewazor/demo/App.java:19: error: incompatible types: String cannot be converted to int
                    System.getenv("HTTP_PORT")
                                 ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
1 actionable task: 1 executed
```

It looks like the `System.getenv` call has the wrong return type here - the string needs to be parsed into an int. I would also suggest adding a default value here, so that in the event that the environment variable is not set or contains an empty string the server can still start up with the default 8080 port number. This is already generally handled by the entrypoint.sh change, but it wouldn't hurt to be defensive here and still make that check.